system/class/Database/Database.php

```diff
@@ -606,7 +610,7 @@ static function changesetMapToList(array $changesetMap): array
 
         foreach ($ids as $id) {
             foreach ($changesetMap[$id] as $column => $value) {
-                $commonChanges[$column][$value][$id] = true;
+                $commonChanges[$column][self::val($value)][$id] = true;
```

Comment:
This should not be needed as boolean keys already get converted to integers. Then the `self::raw()` calls also become unnecessary.